runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -119,6 +117,80 @@ public static String toReadableId(UUID id) {
         return Long.toHexString((id.getLeastSignificantBits()) & 0xFFFF);
     }
 
+
+    /**
+     * Verify that the stripes are not empty and that each segment has one stripe.
+     * @param segments segments to validate
+     */
+    private static void validateSegments(List<Layout.LayoutSegment> segments) {
+        checkNotNull(segments);
+        checkArgument(!segments.isEmpty());
+        long previousSegmentEndAddress = 0;
+        for (Layout.LayoutSegment segment : segments) {
+            checkState(segment.getStart() == previousSegmentEndAddress);
```

Comment:
shouldnt this be segment.getStart() == previousSegmentAddress + 1 ?

Reply:
I don't think so.
Consider the case of one segment [0, -1) , then previousSegmentEndAddress=0 is equal to segment.getStart() which is 0.
Now consider two segments [0,100) [100, -1) , the check is also true here. When we split a segment, the end address of the segment is **non-inclusive** . If they were inclusive, then we need to check with +1 , but they are not.